src/databricks/sql/client.py

```diff
@@ -458,8 +501,36 @@ def _prepare_inline_parameters(
 
         return rendered_statement, NO_NATIVE_PARAMS
 
+    def _prepare_dbsql_params_from_list(
+        self, params: ListOfParameters
+    ) -> List[DbsqlParameter]:
+        """Return a list of DbsqlParameter objects from the passed list of params"""
+
+        output = []
+        for p in params:
+            if isinstance(p, DbsqlParameter):
+                output.append(p)
+            else:
+                output.append(DbsqlParameter(value=p))
+
+        return output
+
+    def _prepare_dbsql_params_from_dict(
+        self, params: DictOfParameters
+    ) -> List[DbsqlParameter]:
+        """Return a  list of DbsqlParameter objects from the passed dictionary"""
+
+        output = []
```

Comment:
As has probably become evident from my other comment, I prefer comprehensions so long as they're not nested.  They make it easier to think/process functionally, because instead of a step by step imperative process that builds up a list with side effects, you can think of the comprehension as a one step data transformation:

```
[DbsqlParameter(name=name, value=value) for (name,value) in params.items()]
```